packages/contract/src/scripts.ts

```diff
@@ -9,35 +12,56 @@ import { Script, ReceiptType } from '@fuel-ts/providers';
  * Accepts a contract ID and function data
  * Returns function result
  */
-export const contractCallScript = new Script<[contractId: BytesLike, data: BytesLike], Uint8Array>(
+export const contractCallScript = new Script<
+  { contractId: BytesLike; assetId?: BytesLike; amount?: BigNumberish; data: BytesLike },
+  Uint8Array
+>(
   /*
-    Opcode::ADDI(0x10, REG_ZERO, script_data_offset)
-    Opcode::CALL(0x10, REG_ZERO, 0x10, REG_CGAS)
-    Opcode::RET(REG_RET)
-    Opcode::NOOP
+    // Load call data to 0x10.
+    Opcode::ADDI(0x10, REG_ZERO, data_offset + 40),
+    // Load gas forward to 0x11.
+    // Load the amount into 0x12
+    Opcode::ADDI(0x12, REG_ZERO, data_offset + 32),
+    // Load word into 0x12
+    Opcode::LW(0x12, 0x12, 0),
+    // Load the asset id to use to 0x13.
+    Opcode::ADDI(0x13, REG_ZERO, data_offset),
+    // Call the transfer contract.
+    Opcode::CALL(0x10, 0x12, 0x13, REG_CGAS),
+    Opcode::RET(REG_ONE),
   */
```

Comment:
Is this correct? A comment says `Load gas forward to 0x11`, but I don't see `0x11` being used.

Reply:
This is a good point, during don't correctly the answer. cc @digorithm.

Reply:
Oh, I forgot to remove that comment on the Rust SDK I guess? We're still not forwarding gas. Once we do we can use 0x11 (or any unreserved) for that.